src/main/java/uk/gov/companieshouse/insolvency/delta/processor/InsolvencyDeltaProcessor.java

```diff
@@ -47,59 +48,66 @@ public InsolvencyDeltaProcessor(ApiClientService apiClientService,
      * Process CHS Delta message.
      */
     public void processDelta(Message<ChsDelta> chsDelta) {
-        try {
-            MessageHeaders headers = chsDelta.getHeaders();
-            final ChsDelta payload = chsDelta.getPayload();
-            final String logContext = payload.getContextId();
-            final Map<String, Object> logMap = new HashMap<>();
-            final String receivedTopic =
-                    Objects.requireNonNull(headers.get(KafkaHeaders.RECEIVED_TOPIC)).toString();
-            final String partition =
-                    Objects.requireNonNull(
-                            headers.get(KafkaHeaders.RECEIVED_PARTITION_ID)
-                    ).toString();
-            final String offset =
-                    Objects.requireNonNull(headers.get(KafkaHeaders.OFFSET)).toString();
-
-            ObjectMapper mapper = new ObjectMapper();
-            InsolvencyDelta insolvencyDelta = mapper.readValue(payload.getData(),
-                    InsolvencyDelta.class);
-            logger.trace(String.format("DSND-362: InsolvencyDelta extracted "
+        MessageHeaders headers = chsDelta.getHeaders();
+        final ChsDelta payload = chsDelta.getPayload();
+        final String logContext = payload.getContextId();
+        final Map<String, Object> logMap = new HashMap<>();
+        final String receivedTopic =
+                Objects.requireNonNull(headers.get(KafkaHeaders.RECEIVED_TOPIC)).toString();
+        final String partition =
```

Comment:
Topic, partition and offset are part of the header of the receiveMainMessages method.
We can probably pass them to this method instead than retrieve them from the KafkaHeaders object.